Add documentation of data logs and the data log tool

PeterJohnson opened the pull request.

PeterJohnson commented on source/docs/software/telemetry/telemetry.rst:
```diff
@@ -25,3 +27,225 @@ WPILib supports several :ref:`dashboards <docs/software/dashboards/index:Dashboa
 
 While NetworkTables does not yet support serialization of complex data types (this is tentatively scheduled for 2023), *mutable* types from user code can be easily extended to interface directly with WPILib dashboards via the ``Sendable`` interface, whose usage is described in the next article.
+
+On-Robot Telemetry Recording Into Data Logs
```
Done.

Daltz333 reviewed (commented):
I love this and I love the documentation here. One small note, all the references to button labels should be formatted like `:guilabel:'mylabel'` where you replace the apostrophe with `

Daltz333 commented on source/docs/software/telemetry/datalog.rst:
```diff
@@ -0,0 +1,221 @@
+
+Log files are initially named ``FRC_TBD_{random}.wpilog`` until the DS connects.  After the DS connects, the log file is renamed to ``FRC_yyyyMMdd_HHmmss.wpilog`` (where the date/time is UTC).  If the FMS is connected and provides a match number, the log file is renamed to ``FRC_yyyyMMdd_HHmmss_{event}_{match}.wpilog``.
+
+On startup, all existing ``FRC_TBD`` log files are deleted.  If there is less than 50 MB of free space on the target storage, ``FRC_`` log files are deleted (oldest to newest) until there is 50 MB free OR there are 10 files remaining.
```
This is slightly confusing? Perhaps rewording this sentence to 
```suggestion
On startup, all existing log files where a DS has not been connected will be deleted.  If there is less than 50 MB of free space on the target storage, ``FRC_`` log files are deleted (oldest to newest) until there is 50 MB free OR there are 10 files remaining.
```

Daltz333 commented on source/docs/software/telemetry/datalog.rst:
```diff
@@ -0,0 +1,221 @@
+
+The ``DataLog`` class (`Java <https://first.wpi.edu/wpilib/allwpilib/docs/release/java/edu/wpi/util/datalog/DataLog.html>`__, `C++ <https://first.wpi.edu/wpilib/allwpilib/docs/release/cpp/classwpi_1_1log_1_1_data_log.html>`__) and its associated LogEntry classes (e.g. ``BooleanLogEntry``, ``DoubleLogEntry``, etc) provides low-level access for writing data logs.
+
+.. note: Unlike NetworkTables, there is no change checking performed.  **Every** call to a LogEntry.append() function will result in a record being written to the data log.  Checking for changes and only appending to the log when necessary is the responsibility of the caller.
```
```suggestion
.. note: Unlike NetworkTables, there is no change checking performed.  **Every** call to a ``LogEntry.append()`` function will result in a record being written to the data log.  Checking for changes and only appending to the log when necessary is the responsibility of the caller.
```

PeterJohnson commented:
Addressed comments.  The links to first.wpi.edu docs still need to be checked; the 2022.4.1 docs haven't published yet so it's uncertain if they're correct.

sciencewhiz commented on source/docs/software/telemetry/datalog.rst:
```diff
@@ -0,0 +1,221 @@
+Standard Data Logging using DataLogManager
+------------------------------------------
+
+The ``DataLogManager`` class (`Java <https://first.wpi.edu/wpilib/allwpilib/docs/release/java/edu/wpi/first/wpilibj/DataLogManager.html>`__, `C++ <https://first.wpi.edu/wpilib/allwpilib/docs/release/cpp/classfrc_1_1_data_log_manager.html>`__) provides a centralized data log that provides automatic data log file management.  It automatically cleans up old files when disk space is low and renames t ...
```
Should we say a fat32 formatted flash drive?

Daltz333 replied:
Does exfat not also work?

sciencewhiz replied:
I was primarily meaning to exclude NTFS. I haven't tried exfat

sciencewhiz replied:
It appears that exfat doesn't get mounted

PeterJohnson replied:
Added a note box regarding USB flash drive formatting requirements.
